### PR TITLE
Current Develop

### PR DESCRIPTION
**Pull request by talihomz**
Here is our latest stable code for Fakebook.

### CODE REVIEW

File: Dockerfile
Comment (nidalaa) [inline Dockerfile:1]: ❤️

File: app/controllers/posts_controller.rb
Comment (nidalaa) [inline app/controllers/posts_controller.rb:43]: It looks like you used scaffold option. That's good but let's make sure that we maintain code that we really use: do you use json format anywhere or plan to do so?

File: app/controllers/posts_controller.rb
Comment (nidalaa) [inline app/controllers/posts_controller.rb:60]:
```
def post
 @post ||= Post.find(params[:id])
end```

will look better than callback

File: app/controllers/posts_controller.rb
Comment (nidalaa) [inline app/controllers/posts_controller.rb:66]: why don't you whitelist anything?

File: app/views/layouts/_navbar.html.erb
Comment (nidalaa) [inline app/views/layouts/_navbar.html.erb:14]: Instead of using html tags inside string + html_safe, you can just use erb functions:
```
<a href=<%= notifications_path %> >
            <i class="material-icons">notifications</i>
           </a>
```

File: app/controllers/posts_controller.rb
Comment (nidalaa) [reply app/controllers/posts_controller.rb:66]: forget it, j just prepared for future, I get it

File: app/views/layouts/_navbar.html.erb
Comment (nidalaa) [inline app/views/layouts/_navbar.html.erb:28]: What is the reason of naming user a resource? Resource is very generic name....
Why don't you just use User.new directly?

File: app/views/notification/index.html.erb
Comment (nidalaa) [inline app/views/notification/index.html.erb:1]: Nice lorem ipsum :D

File: config/routes.rb
Comment (nidalaa) [inline config/routes.rb:18]: Home#welcome seems to be a log in page 🤔 

Why not to change app/views/devise/session/new.html  and use it here?

Comment (nidalaa) [reply config/routes.rb:18]: http://localhost:3000/users/sign_in works ;)

File: app/views/layouts/_navbar.html.erb
Comment (nidalaa) [inline app/views/layouts/_navbar.html.erb:2]: You might want to use slim or haml for your views. It is nicer than erb :)

**Comment by nidalaa** [comment]
Hi @talihomz and @SavaVuckovic  a quick emergency hangout with Kevin helped me understand how you are working. This is the recap:

- master branch is supposed to contain changes that are released to production (therefore, first changed will land there when we have a MVP version of the app)
- develop branch is the main branch
- all other branches are feature-branches, every feature-branch is initiated from develop branch

**Changes to your current workflow:**

- once feature-branch is read a new Github PR to develop branch is opened, based on changes between feature-branch and develop (e.g. https://github.com/talihomz/FakeBook/compare/develop...feature/ui-theme)
- PR is reviewed by me (changes required or not) and merged to develop
- it would be worth to set develop as default branch here https://github.com/talihomz/FakeBook/settings/branches
- and maybe even `Require pull request reviews before merging` here: https://github.com/talihomz/FakeBook/settings/branches/master

Thanks to this I will be able to give you some feedback in early stages. Which will be reasonable in the bigger project.

**Current develop branch**

As there are some minor comments to the code that already has landed in develop, I am going to leave my comments here.

**HOWEVER, PLEASE DO NOT MERGE THIS BRANCH TO MASTER**

Instead:

- please initiate a new branch called `issue-1`
- make all necessary changes and create a new Github PR
- I will review the PR
- and it will go to develop

**Review by nidalaa — CHANGES_REQUESTED** [review]

**Comment by nidalaa** [comment]
Hi @talihomz and @SavaVuckovic  a quick emergency hangout with Kevin helped me understand how you are working. This is the recap:

- master branch is supposed to contain changes that are released to production (therefore, first changed will land there when we have a MVP version of the app)
- develop branch is the main branch
- all other branches are feature-branches, every feature-branch is initiated from develop branch

**Changes to your current workflow:**

- once feature-branch is read a new Github PR to develop branch is opened, based on changes between feature-branch and develop (e.g. https://github.com/talihomz/FakeBook/compare/develop...feature/ui-theme)
- PR is reviewed by me (changes required or not) and merged to develop
- it would be worth to set develop as default branch here https://github.com/talihomz/FakeBook/settings/branches
- and maybe even `Require pull request reviews before merging` here: https://github.com/talihomz/FakeBook/settings/branches/master

Thanks to this I will be able to give you some feedback in early stages. Which will be reasonable in the bigger project.

**Current develop branch**

As there are some minor comments to the code that already has landed in develop, I am going to leave my comments here.

**HOWEVER, PLEASE DO NOT MERGE THIS BRANCH TO MASTER**

Instead:

- please initiate a new branch called `issue-1`
- make all necessary changes and create a new Github PR
- I will review the PR
- and it will go to develop

File: app/controllers/posts_controller.rb
Comment (talihomz) [reply app/controllers/posts_controller.rb:43]: Ok. We will clean this out of our codebase. Is it useful for when we just want to expose an API?

File: app/controllers/posts_controller.rb
Comment (talihomz) [reply app/controllers/posts_controller.rb:66]: Coming soon 😉

File: app/views/layouts/_navbar.html.erb
Comment (talihomz) [reply app/views/layouts/_navbar.html.erb:14]: Fixed :-)

File: app/views/layouts/_navbar.html.erb
Comment (talihomz) [reply app/views/layouts/_navbar.html.erb:28]: Fixed :-)

File: app/views/notification/index.html.erb
Comment (talihomz) [reply app/views/notification/index.html.erb:1]: 😂

File: app/views/layouts/_navbar.html.erb
Comment (talihomz) [reply app/views/layouts/_navbar.html.erb:2]: We wanted to follow Facebook's example of having the signup/login page as the home page. This means we had to change a few default routes to match this specific case.

File: app/controllers/posts_controller.rb
Comment (nidalaa) [reply app/controllers/posts_controller.rb:43]: Yes and it could be either API for external clients or for your app as well (e.g. if your views are written with some frontend framework).

File: config/routes.rb
Comment (nidalaa) [reply config/routes.rb:18]: I guess  this comment https://github.com/talihomz/FakeBook/pull/3/files/d77418cd7b96494ff3f409df2db4960ae8d43cdb#r187860074 
was about this line of code.

>We wanted to follow Facebook's example of having the signup/login page as the home page. This means we had to change a few default routes to match this specific case.

You mean, you want  http://localhost:3000 (or www.fakebook.com) to
- open signin/up form for unauthenticated users
- open home index for authenticated users,

right?

Still my question is valid? Why do you need to create new action `home#welcome` instead of using `devise/session#new` ?

Comment (nidalaa) [reply config/routes.rb:18]: @talihomz @SavaVuckovic ☝️ check `rake routes` to see what devise is adding for you :)

Comment (SavaVuckovic) [reply config/routes.rb:18]: We have fixed this :)

**Pull request closed without merging**
